lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -399,21 +440,32 @@ def test_http_exception(
                 self.date_range(self.payload["start"], self.payload["end"]),
             )
 
-        query_api(
-            self.pbench_endpoint,
-            self.elastic_endpoint,
-            self.payload,
-            index,
-            exceptions["status"],
-            body=exceptions["exception"],
-            headers={"Authorization": "Bearer " + pbench_token},
-        )
+        if self.api_method == API_METHOD.GET:
+            query_api_get(
+                self.pbench_endpoint,
+                self.elastic_endpoint,
+                index,
+                exceptions["status"],
+                body=exceptions["exception"],
+                headers={"Authorization": "Bearer " + pbench_token},
+            )
+        elif self.api_method == API_METHOD.POST:
+            query_api(
+                self.pbench_endpoint,
+                self.elastic_endpoint,
+                self.payload,
```

Comment:
Can we make this an optional parameter, so we don't have to have this duplicate code?

Reply:
Peter, is the addition of `query_api_get()` the duplication that you're trying to avoid?

Yeah, it looks like that can be done with an optional parameter, but I think it's going to make a complicated piece of logic even more complex (I'm too foggy right now to figure out how much).

Reply:
Note that `self.payload` on your new `GET` API classes will be `None`, so a single pattern ought to work with my suggested consolidated `query_api`. (Though "some assembly may be required" 😆)